test/controllers/CreateOrAmendInterestControllerSpec.scala

```diff
@@ -0,0 +1,101 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import connectors.httpParsers.CreateOrAmendInterestHttpParser.CreateOrAmendInterestResponse
+import models.{CreateOrAmendInterestModel, NotFoundError}
+import org.scalamock.handlers.CallHandler4
+import play.api.http.Status._
+import play.api.libs.json.Json
+import services.CreateOrAmendInterestService
+import testUtils.TestSuite
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.Future
+
+class CreateOrAmendInterestControllerSpec extends TestSuite {
+
+  val serviceMock: CreateOrAmendInterestService = mock[CreateOrAmendInterestService]
+  val controller = new CreateOrAmendInterestController(serviceMock, mockControllerComponents, authorisedAction)
+
+  val nino = "nino"
+  val taxYear = 2021
+  val incomeSourceName = "incomeSourceNameTest"
+  val incomeSourceId = "incomeSourceIdTest"
+  val mtditid = "someMtditid"
+
+  val interestSubmittedModel: Seq[CreateOrAmendInterestModel] = Seq(CreateOrAmendInterestModel(Some(incomeSourceId), incomeSourceName, None, Some(100.00)))
+  val interestSubmittedModelInvalid: CreateOrAmendInterestModel = CreateOrAmendInterestModel(Some(incomeSourceId), incomeSourceName, None, Some(100.00))
+
+  val interestSuccessResponse: Seq[Future[CreateOrAmendInterestResponse]] =
+    Seq(Future.successful(Right(true)), Future.successful(Right(true)), Future.successful(Right(true)))
+
+  val interestFailResponse: Seq[Future[CreateOrAmendInterestResponse]] =
+    Seq(Future.successful(Right(true)), Future.successful(Left(NotFoundError)), Future.successful(Right(true)))
+
+  def mockServiceSuccessCall: CallHandler4[String, Int, Seq[CreateOrAmendInterestModel], HeaderCarrier, Seq[Future[CreateOrAmendInterestResponse]]] =
+    (serviceMock.createOrAmendAllInterest(_: String, _: Int, _: Seq[CreateOrAmendInterestModel])(_: HeaderCarrier))
+      .expects(nino, taxYear, interestSubmittedModel, *)
+      .returning(interestSuccessResponse)
+
+  def mockServiceFailCall: CallHandler4[String, Int, Seq[CreateOrAmendInterestModel], HeaderCarrier, Seq[Future[CreateOrAmendInterestResponse]]] =
+    (serviceMock.createOrAmendAllInterest(_: String, _: Int, _: Seq[CreateOrAmendInterestModel])(_: HeaderCarrier))
+      .expects(nino, taxYear, interestSubmittedModel, *)
+      .returning(interestFailResponse)
+
+  ".createOrAmendInterest" should {
+    "return a no content" when {
+    "passed a validModel and there are no failures from the connector" in {
+
+      val expectedResult = NO_CONTENT
+
+      mockAuth()
+      mockServiceSuccessCall
+      val result = controller.createOrAmendInterest(nino, taxYear, mtditid)(fakeRequestWithMtditid.withJsonBody(Json.toJson(interestSubmittedModel)))
+
+
+      status(result) mustBe expectedResult
+
+    }
+  }
+    "return a badRequest" when {
```

Comment:
Should be returns an error, not returns a badRequest specifically, as one of the tests below returns an INTERNAL_SERVER_ERROR